### ISSUE TITLE
ComplexHalf Coverage Tracker

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#74537]
### 🚀 The feature, motivation and pitch

There have been multiple requests and issues (see https://github.com/pytorch/pytorch/issues/71680) to have better operator support for `complex32`. Currently it is almost impossible to do anything with a `Tensor` of `complex32` dtype (even printing doesn't work).

All operators which support complex dtypes should ideally support `complex32`. Computation will occur in `complex64` similar to `Half` wherein computation occurs in `float`.

Following is the operator coverage:

Fundamental Ops (expected 1.12)
* [ ] copy_ (to support casting to and from `complex32`)
* [ ] print support (Ref: https://github.com/pytorch/pytorch/issues/50097)

**NOTE**: List will be updated if supporting above list requires supporting another operator 

High-Priority Ops (based on user demand)
* [ ] Support for fft module (expected 1.12)
    * [ ] fft.fft
    * [ ] fft.fft2
    * [ ] fft.fftn
    * [ ] fft.fftshift
    * [ ] fft.hfft
    * [ ] fft.hfft2
    * [ ] fft.hfftn
    * [ ] fft.ifft
    * [ ] fft.ifft2
    * [ ] fft.ifftn
    * [ ] fft.ifftshift
    * [ ] fft.irfft
    * [ ] fft.irfft2
    * [ ] fft.irfftn
* [ ] Support for stft (see #67324) (expected 1.12)
    * [ ] fill_
* [ ] Support for convolution (see also #71108)

High-Priority Ops (expected 1.12)
* [ ] add
* [ ] sub
* [ ] mul
* [ ] div
* [ ] sum
* [ ] mean
* [ ] equal
* [ ] imag
* [ ] real
* [ ] H
* [ ] T
* [ ] \_\_getitem\_\_
* [ ] Support testing with OpInfo infra
    * [ ] testing.make_tensor
    * [ ] complex (required for make_tensor)

Tensor Creation Ops:
* [ ] randn (supported, needs tests)
* [ ] rand (supported, needs tests)
* [ ] empty (supported, needs tests)
* [ ] full (supported only on CPU, needs tests)
* [ ] ones (supported only on CPU, needs tests)
* [ ] zeros (supported only on CPU, needs tests)
* [ ] linspace
* [ ] logspace
* [ ] arange
* [ ] eye

Remaining Operators which support complex type (generated from OpInfo)
* [ ] \_\_radd\_\_
* [ ] \_\_rdiv\_\_
* [ ] \_\_rmatmul\_\_
* [ ] \_\_rmul\_\_
* [ ] \_\_rpow\_\_
* [ ] \_\_rsub\_\_
* [ ] _masked.mean
* [ ] _masked.prod
* [ ] _masked.sum
* [ ] _masked.var
* [ ] abs
* [ ] acos
* [ ] acosh
* [ ] addbmm
* [ ] addcdiv
* [ ] addcmul
* [ ] addmm
* [ ] addmv
* [ ] addr
* [ ] allclose
* [ ] angle
* [ ] any
* [ ] argwhere
* [ ] as_strided
* [ ] asin
* [ ] asinh
* [ ] atan
* [ ] atanh
* [ ] atleast_1d
* [ ] atleast_2d
* [ ] atleast_3d
* [ ] baddbmm
* [ ] bfloat16
* [ ] block_diag
* [ ] bmm
* [ ] bool
* [ ] broadcast_tensors
* [ ] broadcast_to
* [ ] byte
* [ ] cartesian_prod
* [ ] cat
* [ ] char
* [ ] cholesky
* [ ] cholesky_inverse
* [ ] cholesky_solve
* [ ] chunk
* [ ] clone
* [ ] column_stack
* [ ] combinations
* [ ] conj
* [ ] conj_physical
* [ ] contiguous
* [ ] corrcoef
* [ ] cos
* [ ] cosh
* [ ] count_nonzero
* [ ] cov
* [ ] cross
* [ ] cumprod
* [ ] cumsum
* [ ] cumulative_trapezoid
* [ ] diag
* [ ] diag_embed
* [ ] diagflat
* [ ] diagonal
* [ ] diff
* [ ] dist
* [ ] dot
* [ ] double
* [ ] dsplit
* [ ] dstack
* [ ] eig
* [ ] einsum
* [ ] empty_like
* [ ] eq
* [ ] exp
* [ ] expand
* [ ] expand_as
* [ ] flatten
* [ ] flip
* [ ] fliplr
* [ ] flipud
* [ ] float
* [ ] float_power
* [ ] full_like
* [ ] gather
* [ ] geqrf
* [ ] gradient
* [ ] half
* [ ] hsplit
* [ ] hstack
* [ ] index_add
* [ ] index_copy
* [ ] index_fill
* [ ] index_put
* [ ] index_select
* [ ] inner
* [ ] int
* [ ] inverse
* [ ] isclose
* [ ] isfinite
* [ ] isinf
* [ ] isnan
* [ ] isreal
* [ ] istft
* [ ] kron
* [ ] ldexp
* [ ] lerp
* [ ] linalg.cholesky
* [ ] linalg.cholesky_ex
* [ ] linalg.cond
* [ ] linalg.cross
* [ ] linalg.det
* [ ] linalg.eig
* [ ] linalg.eigh
* [ ] linalg.eigvals
* [ ] linalg.eigvalsh
* [ ] linalg.householder_product
* [ ] linalg.inv
* [ ] linalg.inv_ex
* [ ] linalg.lstsq
* [ ] linalg.lu_factor
* [ ] linalg.lu_factor_ex
* [ ] linalg.matrix_norm
* [ ] linalg.matrix_power
* [ ] linalg.matrix_rank
* [ ] linalg.multi_dot
* [ ] linalg.norm
* [ ] linalg.pinv
* [ ] linalg.qr
* [ ] linalg.slogdet
* [ ] linalg.solve
* [ ] linalg.solve_triangular
* [ ] linalg.svd
* [ ] linalg.svdvals
* [ ] linalg.tensorinv
* [ ] linalg.tensorsolve
* [ ] linalg.vector_norm
* [ ] log
* [ ] log10
* [ ] log2
* [ ] log_softmax
* [ ] logical_and
* [ ] logical_not
* [ ] logical_or
* [ ] logical_xor
* [ ] long
* [ ] lu
* [ ] lu_solve
* [ ] lu_unpack
* [ ] mH
* [ ] mT
* [ ] masked_fill
* [ ] masked_scatter
* [ ] masked_select
* [ ] matmul
* [ ] matrix_exp
* [ ] meshgrid
* [ ] mm
* [ ] movedim
* [ ] mv
* [ ] narrow
* [ ] ne
* [ ] neg
* [ ] new_empty
* [ ] new_full
* [ ] new_ones
* [ ] new_zeros
* [ ] nn.functional.feature_alpha_dropout
* [ ] nn.functional.linear
* [ ] nn.functional.normalize
* [ ] nn.functional.pad
* [ ] nn.functional.pairwise_distance
* [ ] nn.functional.pixel_shuffle
* [ ] nn.functional.pixel_unshuffle
* [ ] nn.functional.silu
* [ ] nn.functional.softmin
* [ ] nn.functional.softsign
* [ ] nn.functional.tanhshrink
* [ ] nn.functional.unfold
* [ ] nonzero
* [ ] norm
* [ ] ones_like
* [ ] ormqr
* [ ] outer
* [ ] permute
* [ ] pinverse
* [ ] positive
* [ ] pow
* [ ] prod
* [ ] put
* [ ] qr
* [ ] rand_like
* [ ] randn_like
* [ ] ravel
* [ ] reciprocal
* [ ] renorm
* [ ] repeat
* [ ] repeat_interleave
* [ ] reshape
* [ ] reshape_as
* [ ] resize_
* [ ] resize_as_
* [ ] resolve_conj
* [ ] resolve_neg
* [ ] roll
* [ ] rot90
* [ ] rsqrt
* [ ] rsub
* [ ] scatter
* [ ] scatter_add
* [ ] select
* [ ] sgn
* [ ] short
* [ ] sigmoid
* [ ] sin
* [ ] sinc
* [ ] sinh
* [ ] softmax
* [ ] solve
* [ ] split
* [ ] split_with_sizes
* [ ] sqrt
* [ ] square
* [ ] squeeze
* [ ] stack
* [ ] std
* [ ] std_mean
* [ ] sum_to_size
* [ ] svd
* [ ] symeig
* [ ] t
* [ ] take
* [ ] take_along_dim
* [ ] tan
* [ ] tanh
* [ ] tensor_split
* [ ] tensordot
* [ ] tile
* [ ] to_sparse
* [ ] trace
* [ ] transpose
* [ ] trapezoid
* [ ] trapz
* [ ] triangular_solve
* [ ] tril
* [ ] triu
* [ ] true_divide
* [ ] unfold
* [ ] unsqueeze
* [ ] var
* [ ] var_mean
* [ ] vdot
* [ ] view
* [ ] view_as
* [ ] view_as_real
* [ ] vsplit
* [ ] vstack
* [ ] where
* [ ] zero_
* [ ] zeros_like

cc: @anjal

**Comment by bhack** [comment]
Is this still updated?

**Comment by kshitij12345** [comment]
Unfortunately, I don't think so.

**Comment by bhack** [comment]
So, can we close it?
